BUILDING.md

```diff
@@ -0,0 +1,57 @@
+# How to Compile
+
+FileOnQ.Imaging.Raw uses a mix of .NET Framework, .NET, native C/C++ projects, and CUDA kernels for GPU processing. You will need to have additional build tools installed to properly build this locally. This readme serves as a guide for new developers to contribute.
+
+## Visual Studio and .NET
+
+This project uses a combination of Visual Studio and .NET CLI tools for building and developing.
+
+* Visual Studio 2019 (16.10.3) or newer
+  * Workload: Desktop development with C++
+  * Component: Windows 10 SDK (10.0.18)
+* .NET CLI 5.0.300 or newer
+
+If you are using an older version of Visual Studio or .NET CLI you won't have access to the `_OR_GREATER` preprocessor directives. This will result in runtime issues when running the library locally.
+
+## Build Dependencies
+
+* LibRaw: add these dependencies by running the command `git submodule update --init --recursive` in the repo's root folder. or pull them with the repo using `git clone --recurse-submodules https://github.com/FileOnQ.Imaging.Raw.git`
+* CUDA: You will need to download and install [CUDA SDK 11.4](https://developer.nvidia.com/cuda-11-4-3-download-archive)
+
+## Compile
+
+First time compilation takes a while. Future builds won't take as long even if you do a clean and rebuild.
+
+By compiling the main project `FileOnQ.Imaging.Raw` you will be compiling all native dependencies. No need to run any additional compilation instructions.
+
+### Visual Studio
+
+Right click on `FileOnQ.Imaging.Raw` and select build
+
+### CLI
+
+```shell
+dotnet build src/FileOnQ.Imaging.Raw/FileOnQ.Imaging.Raw.csproj
+```
+
+### Troubleshooting
+
+If after doing all of the above and compiling is causing errors. Try opening a developer command prompt as admin, navigate to the repo's root directory and run `msbuild /p:Platform="x64"` or `msbuild /p:Platform="x86"`
```

Comment:
When adding terminal commands I prefer to have them in a code block instead of inline. It is easier to read for folx that are skimming the docs.

--
If after doing all of the above and compiling is causing errors. Try opening a developer command prompt as admin, navigate to the repo's root directory and run
```
msbuild /p:Platform="x64"
```
or
```
msbuild /p:Platform="x86"
```